cwms-data-api/src/main/java/cwms/cda/ApiServlet.java

```diff
@@ -313,59 +314,98 @@ protected void configureRoutes() {
+                new TimeSeriesGroupController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/timeseries/{timeseries}", tsController, requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/ratings/template/{template-id}",
+                new RatingTemplateController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/ratings/spec/{rating-id}",
+                new RatingSpecController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/ratings/metadata/{rating-id}",
+                new RatingMetadataController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/ratings/{rating-id}",
+                new RatingController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/catalog/{dataset}",
+                new CatalogController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/basins/{basin-id}",
+                new BasinController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/blobs/{blob-id}",
+                new BlobController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/clobs/{clob-id}",
+                new ClobController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/pools/{pool-id}",
+                new PoolController(metrics), requiredRoles,5, TimeUnit.MINUTES);
+        cdaCrudCache("/specified-levels/{specified-level-id}",
+                new SpecifiedLevelController(metrics), requiredRoles,5, TimeUnit.MINUTES);
     }
 
+    /**
+     * This method delegates to the cdaCrud method but also adds an after filter for the specified
+     * path.  If the request was a GET request and the response does not already include
+     * Cache-Control then the filter will add the Cache-Control max-age header with the specified
+     * number of seconds.
+     * Controllers can include their own Cache-Control headers via:
+     *  "ctx.header(Header.CACHE_CONTROL, " public, max-age=" + 60);"
+     * This method lets the ApiServlet configure a default max-age for controllers that don't or
+     * forget to set their own.
+     * @param path where to register the routes.
+     * @param crudHandler the handler requests should be forwarded to.
+     * @param roles the required these roles are present to access post, patch
+     * @param duration the number of TimeUnit to cache GET responses.
+     * @param timeUnit the TimeUnit to use for duration.
+     */
+    public static void cdaCrudCache(@NotNull String path, @NotNull CrudHandler crudHandler,
+                                    @NotNull RouteRole[] roles, long duration, TimeUnit timeUnit) {
+        cdaCrud(path, crudHandler, roles);
+        addCacheControl(path, duration, timeUnit);
+    }
+
+    private static void addCacheControl(@NotNull String path, long duration, TimeUnit timeUnit) {
+        if(timeUnit != null && duration > 0) {
+            staticInstance().after(path, ctx -> {
```

Comment:
This is matching on the exact string passed in. in the case of states it's waiting for exactly `/states/{state}`

This worked: `... after(path.replaceAll("(.*)(\\{.*\\})","$1*"), ...`

Reply:
States was maybe a bad controller to test this with b/c States doesn't have a getOne method.   Having the replacement string in the path does work for matching requests that include the resourceId but it doesn't match the getAll style requests.  I played around with the regex a bit but I couldnt' get it to work to match both types of url so I just add two after handlers.

Reply:
I would say it was a good controller as it exposed a problem.

Reply:
oh, it should be `(.*)(\\{.*\\})?` so that the path parameter is optional... though that wouldn't match if we include multiple path parameters so maybe a replaceAll with just the `\\{.*\\}` -> `*` would do the trick.

Reply:
I added a test where you can play around with the regex approach.  I couldn't get it to work - from what I can see the after handler system doesn't use regex - they do their own custom matching and manipulation.

Reply:
the regex isn't used by the after handler, it just substitutes an "{text}" with with * so "/locations/{location-id}" becomes "/locations/*" as "/path/*" was the example shown.

Though I can except that that still isn't quite enough, the example was limited and some of our handler path can be a bit more complex.